src/commands/run.rs

```diff
@@ -0,0 +1,263 @@
+    #[arg(long)]
+    pub no_color: bool,
+
+    /// The report mode.
+    #[arg(short = 'm', long, default_value_t, value_name = "MODE")]
+    pub report_mode: Mode,
+}
+
+/// Creates the output directory for the task.
+fn create_output_dir(output_dir: Option<PathBuf>, name: &str, overwrite: bool) -> Result<PathBuf> {
+    let output_dir = output_dir.unwrap_or_else(|| {
+        let timestamp = chrono::Utc::now().format("%Y%m%d_%H-%M-%S");
+        PathBuf::from(format!("sprocket_runs/{}/{}", name, timestamp))
+    });
+
+    let output_dir = if output_dir.exists() {
+        if overwrite {
+            fs::remove_dir_all(&output_dir).with_context(|| {
+                format!(
+                    "failed to remove output directory `{dir}`",
+                    dir = output_dir.display()
+                )
+            })?;
+            output_dir
+        } else {
+            bail!(
+                "output directory `{dir}` already exists; use the `--overwrite` option to \
+                 overwrite it",
+                dir = output_dir.display()
+            );
+        }
+    } else {
+        output_dir
+    };
+
+    fs::create_dir_all(&output_dir).with_context(|| {
+        format!(
+            "failed to create output directory `{dir}`",
+            dir = output_dir.display()
+        )
+    })?;
+
+    log::info!(
+        "output directory created: `{dir}`",
+        dir = output_dir.display()
+    );
+
+    Ok(output_dir)
+}
+
+/// Runs a task.
+pub async fn run(args: RunArgs) -> Result<()> {
+    eprintln!(
+        "the `run` command is in alpha testing and does not currently support workflows or using \
+         containers."
+    );
+
+    let file = args.file;
+    if Path::new(&file).is_dir() {
+        anyhow::bail!("expected a WDL document, found a directory");
+    }
+    let (config, mut stream) = get_display_config(args.report_mode, args.no_color);
+
+    let results = analyze(&file, vec![], false, false).await?;
+
+    let uri = if let Ok(uri) = Url::parse(&file) {
+        uri
+    } else {
+        path_to_uri(&file).expect("file should be a local path")
+    };
+
+    let result = results
+        .iter()
+        .find(|r| **r.document().uri() == uri)
+        .context("failed to find document in analysis results")?;
+    let document = result.document();
+
+    let mut engine = Engine::new(LocalTaskExecutionBackend::new());
+    let (path, name, inputs) = if let Some(path) = args.inputs {
+        let abs_path = absolute(&path).with_context(|| {
+            format!(
+                "failed to determine the absolute path of `{path}`",
+                path = path.display()
+            )
+        })?;
+        match Inputs::parse(document, &abs_path)? {
+            Some((name, inputs)) => (Some(path), name, inputs),
+            None => bail!(
+                "inputs file `{path}` is empty; use the `--name` option to specify the name of \
+                 the task or workflow to run",
```

Comment:
SInce there can be only one `workflow` in a file, is it safe to make an assumption to run that if `name` isn't specified? That's what the other runners do correct?

Reply:
Right now, `wdl-engine` expects the name of the task _or_ workflow to be specified in the JSON (in the form `"<name of task or workflow>.<name of parameter>": <value>`). I played around with trying to loosen that requirement within the `sprocket` bin, but I think it will require some tweaks upstream.

So if no input JSON is supplied, we currently rely on `--name` to select what task/workflow to run. I suppose it would be fine if both are omitted to run any found workflow by default. I can make that change.

Reply:
However, I don't think it's really a common use-case to omit the input JSON, so I'm not certain how important this is. The input JSON must specify the name of the task or workflow, so supplying `--name` is redundant or leads to contradictions, so isn't allowed at the same time. There aren't many tasks or workflows "in the wild" that you can run without specifying any parameters (which currently must be done via JSON).

Reply:
To clarify: the only time a user can or should be using the `--name` parameter is if they want to run a task (or workflow) without any arguments. This is a pretty niche use case.

However in a future update, we will probably want to allow CL specified parameters, and when we add that we will want to use this `--name` flag to enable that behavior.

I think it's fine to require specificity from users and supply the workflow name if they want to use this feature.